src/framework/standard/command.rs

```diff
@@ -116,6 +123,128 @@ pub struct CommandOptions {
 }
 
+pub struct HelpOptions {
+    /// Suggests a command's name.
+    pub suggestion_text: String,
+    /// If no help is available, this text will be displayed.
+    pub no_help_available_text: String,
+    /// How to use a command, `{usage_embed_label}: {command_name} {args}`
+    pub usage_label: String,
+    /// Actual sample label, `{sample_embed_label}: {command_name} {args}`
+    pub usage_sample_label: String,
+    /// Text labeling ungrouped commands, `{ungrouped_label}: ...`
+    pub ungrouped_label: String,
+    /// Text labeling the start of the description.
+    pub description_label: String,
+    /// Text labeling grouped commands, `{grouped_label} {group_name}: ...`
+    pub grouped_label: String,
+    /// Text labeling a command's alternative names (aliases).
+    pub aliases_label: String,
+    /// Text specifying that a command is only usable in a guild.
+    pub guild_only_text: String,
+    /// Text specifying that a command is only usable in via DM.
+    pub dm_only_text: String,
+    /// Text specifying that a command can be used via DM and in guilds.
+    pub dm_and_guild_text: String,
+    /// Text expressing that a command is available.
+    pub available_text: String,
+    /// Error-message once a command could not be found.
+    /// Output-example (without whitespace between both substitutions: `{command_not_found_text}{command_name}`
+    /// `{command_name}` describes user's input as in: `{prefix}help {command_name}`.
+    pub command_not_found_text: String,
+    /// Explains the user on how to use access a single command's details.
+    pub individual_command_tip: String,
+    /// Explains reasoning behind striked commands, see fields requiring `HelpBehaviour` for further information.
+    /// If `HelpBehaviour::Strike` is unused, this field will evaluate to `None` during creation
+    /// inside of `CreateHelpCommand`.
+    pub striked_commands_tip: Option<String>,
+    /// Announcing a group's prefix as in: {group_prefix} {prefix}.
+    pub group_prefix: String,
+    /// If a user lacks required roles, this will treat how these commands will be displayed.
+    pub lacking_role: HelpBehaviour,
+    /// If a user lacks permissions, this will treat how these commands will be displayed.
+    pub lacking_permissions: HelpBehaviour,
+    /// If a user is using the help-command in a channel where a command is not available,
+    /// this behaviour will be executed.
+    pub wrong_channel: HelpBehaviour,
+    /// Colour help-embed will use upon encountering an error.
+    pub embed_error_colour: Colour,
+    /// Colour help-embed will use if no error occured.
+    pub embed_success_colour: Colour,
+}
+
+pub trait HelpCommand: Send + Sync + 'static {
+    fn execute(&self, &mut Context, &Message, &HelpOptions, HashMap<String, Arc<CommandGroup>>, Args) -> Result<(), Error>;
+
+    fn options(&self) -> Arc<CommandOptions> {
+        Arc::clone(&DEFAULT_OPTIONS)
+    }
+}
+
+impl HelpCommand for Arc<Help> {
```

Comment:
i'd favour for this to be an impl for `Arc<HelpCommand>`, as then anything that implements `HelpCommand` and is inside `Arc` will automatically impl `HelpCommand` as well.